### ISSUE TITLE
ShopGUIPlus sync prices issue

### ISSUE THREAD

**Issue by DebuffPvP** [BG-Software-LLC/SuperiorSkyblock2#522]
### Minecraft's Version

Flamepaper 1.8.8

### Plugin's Version

The dev version #244 

### Describe the bug

The item prices stay when the server restarts
Post restart: https://gyazo.com/0078d4803a89c1d09f1189bae06edd66
After restart: https://gyazo.com/9a9b3cf53d62a4bb4797c972ffaa6f0c

### To Reproduce

Make sure you have the default generated files of the plugin and make sure the database file is previously deleted if you made or delete an island previously as this has to do with that. Create an island then do /is top and you will see nothing since nothing has been placed. Place a block with is level and value then check the value and then break it then check /is top again and you will see it went back to 0. Then restart the server log back in and do /is top and you will see the issue.

**Comment by OmerBenGera** [comment]
Have you stopped spamming my github?

**Comment by DebuffPvP** [comment]
Yes i was just mad at being banned and i was banned with no explanation when i asked why because the point in us is reporting bugs to make the plugin better for all not just me and i completely forgot that you're jewish when i said the jews killed jesus and then when i got banned i realized why you probably got mad at the joke i made and you didn't say anything when i was talking like a normal human being and i get that you're overwhelmed and stuff from tickets and you're not available 24/7 but people have made many requests on the same bugs and such i said how it occured and provided the proof via the ticket and via video but you kept say hey it's the upgrades you have that's why but like i told you many times and provided screenshots you kept insisting as if you just skim the message is what it sounded like to me and i'm sorry for the behaviour and no idc what religion someone is cause if you put all religions in one room all of them are people to me like i said i was just mad and that's it

**Comment by DebuffPvP** [comment]
I know i overdid it and i know it seemed like i was being a dick prior to all this happening when you said in one ticket about me but i simply come here to report bugs and provide all the info nessecary and that's it and on discord i have been trying to help people so we can all benefit as far as being an active community i got nothing to do with nobody at all so if i could possibly be unbanned cause i sincerely apologize and actually made some friends off that discord

**Comment by OmerBenGera** [comment]
I will look into that.
Your ban will not get removed from discord until I change my mind.

**Comment by DebuffPvP** [comment]
So how am i supposed to change your mind?

**Comment by OmerBenGera** [comment]
I can't seem to reproduce the issue, it works as intended for me.
I created a new island, checked /is top, placed and broke a block, checked it again.
Then, made a restart to the server, checked /is top again and it works fine.

Check in your database if the worth bonus is set to anything (also check it using /is admin show)

**Comment by DebuffPvP** [comment]
where's the worth bonus at?

**Comment by OmerBenGera** [comment]
> 
> 
> where's the worth bonus at?

In the database or check it using /is admin show, as I said before.

**Comment by DebuffPvP** [comment]
ok i'll is admin show it to you when i get back next week from cancun mexico

**Comment by DebuffPvP** [comment]
https://gyazo.com/8ccd5ca3dc28f34711dd5a669a9cdaf3

**Comment by DebuffPvP** [comment]
No blocks were placed i put a spawner and removed it then stopped and started the server and did /is admin show

**Comment by OmerBenGera** [comment]
Well, seems like you have a worth bonus.
And the issue still occurs? You still have more levels than you usual have? Also, did you stop your server correctly using /stop, so the plugin can save all data etc?

**Comment by DebuffPvP** [comment]
is there an option to disable worth bonus and yes i correctly did /stop as the console stopped entirely and saved all data

**Comment by OmerBenGera** [comment]
> 
> 
> is there an option to disable worth bonus and yes i correctly did /stop as the console stopped entirely and saved all data. yes the issue still occurs and still have more than usual, hence to the question of bonus worth if there is an option to disabled that then i'll retest.

Don't have schematics offset enabled?

**Comment by OmerBenGera** [comment]
But it won't apply it on your island, it will only be applied to new islands.
You can remove worth bonus for your island using /is admin setbonus

**Comment by DebuffPvP** [comment]
No island offset is enabled though! also so how do you make it so an island has no bonus when it's made or any at all and also what is the bonus used for?

**Comment by OmerBenGera** [comment]
> 
> 
> No island offset is enabled though! also so how do you make it so an island has no bonus when it's made or any at all and also what is the bonus used for?

You have a bonus worth, so it was enabled while you created the island
And the bonus is used to offset values for the schematic

**Comment by DebuffPvP** [comment]
so how can i make all islands start at 0 and only add values only if a block is placed?

**Comment by DebuffPvP** [comment]
https://mega.nz/file/R44HWIgB#AP5K_-TXW3ZmpBp82tUIcQsSVB1CTAIVQ-HgNX2dzFg

i put in the whole superiorskyblock with all the stuff in my superior folder along with my shopguiplus stuff and pricing and all i had loaded was these https://gyazo.com/e21eeb9b8299b878234544a3389d6660 so create an island, place a stone block lets say then do /is top, then break it, stop the server then start it up again and do /is top and it will happen 100% but don't change any of the stuff in the superiorskyblock folder i gave you just so we can see if it's my wrong doing or not but i can't really say it's my wrong doing as of right now since i've seen multiple other reports of this

**Comment by OmerBenGera** [comment]
I am not downloading files from strangers online, sorry.
Try removing SGP and try it with only SSB installed.

**Comment by DebuffPvP** [comment]
So apparently the issue only happens when shopguiplus is hooked in

**Comment by DebuffPvP** [comment]
also if you want i can upload those plugins to a host and give you won't need to download anything if you'd like pebble host is like 2 bucks so it's not much

**Comment by DebuffPvP** [comment]
just created island:
shows the blocks of the island and their values using the buy option
https://gyazo.com/04184554dbe35392e565e941e9ef71f3

the is top shows nothing added
https://gyazo.com/cc83502a4ae346d6f9c516fe8b803058

restarted the server and it showed this in the is top
https://gyazo.com/b4f5d6e805bc423e27af16ffbbeb6de3

is admin show:
https://gyazo.com/32ef08ec9e5d8c909f675b9ae25677cf

It also does it when you create an island and stop the server without placing a block and when you do /is recalc it doesn't show any of the levels prior to restart but when the restart happens and i do /is recalc again it includes what's in the is top

**Comment by OmerBenGera** [comment]
> It also does it when you create an island and stop the server without placing a block and when you do /is recalc it doesn't show any of the levels prior to restart but when the restart happens and i do /is recalc again it includes what's in the is top

If you run /is admin recalc, does your level go to 0 again?

**Comment by DebuffPvP** [comment]
It stays the same

**Comment by OmerBenGera** [comment]
> 
> 
> It stays the same

Do you see any differences in the counts between before and after the restart?

**Comment by DebuffPvP** [comment]
no difference

**Comment by OmerBenGera** [comment]
> 
> 
> no difference

If you run /is admin recalc before the restart, does it change anything?

**Comment by DebuffPvP** [comment]
so apparently the /is recalc set it back to 0 and for some reason in regards to the recalc it sometimes works and sometimes it doesn't

**Comment by OmerBenGera** [comment]
> 
> 
> so apparently the /is recalc set it back to 0 and for some reason in regards to the recalc it sometimes works and sometimes it doesn't

I don't understand why a restart causes it - nothing changes in a restart.
Check the database, look if the counts are different there somehow

**Comment by OmerBenGera** [comment]
Also, you tested it out without SGP, right?

**Comment by DebuffPvP** [comment]
correct and the issue only happens when it hooks into shopguiplus

**Comment by OmerBenGera** [comment]
> 
> 
> correct and the issue only happens when it hooks into shopguiplus

I understand. So to summarize the issue:
There's an issue with prices when they are synced with ShopGUIPlus. After a restart, the worth value of the island is not the same.

**Comment by DebuffPvP** [comment]
So i tried and apparently when sync-worth: NONE is set to none it still happens and i deleted the whole superiorskyblock folder and it also happens with everything as default so long as shopguiplus is present and when i remove shopguiplus and start the server again the same value still stays on the level and worth

**Comment by DebuffPvP** [comment]
i just did is admin recalc and is recalc and it still stayed the same this time

**Comment by DebuffPvP** [comment]
This is the db file https://pastebin.com/ChCPkqH4

**Comment by OmerBenGera** [comment]
So it only happens when SGP is installed, no matter if the prices are synced, right?

**Comment by DebuffPvP** [comment]
So i just tested another method prior to answering this. So when shopguiplus is present and you create an island and stop the server wether shopguiplus is present on the next launch or not that data is still there apparently

**Comment by OmerBenGera** [comment]
I could replicate it only once, which is weird.
I'll look into the code and hopefully I will be able to fix it.

**Comment by DebuffPvP** [comment]
ok finally i'm so glad it happened to you cause me and other people have tried to prove it and i was looking at issues made numbers 483 and 386 and they look similar to my issue somewhat you should try and ask them if they are hooking into shopguiplus to get prices

**Comment by OmerBenGera** [comment]
> 
> 
> ok finally i'm so glad it happened to you cause me and other people have tried to prove it and i was looking at issues made numbers 483 and 386 and they look similar to my issue somewhat you should try and ask them if they are hooking into shopguiplus to get their worths

It seems like the bonus worth was reset to 0 for no reason, that at least what happened to me.

**Comment by DebuffPvP** [comment]
do you think you could possibly fix this issue

**Comment by DebuffPvP** [comment]
any updates on this?

**Comment by OmerBenGera** [comment]
There is no need to bump the issue, and I will get into that once I have the time.
More information about how to replicate it is more than welcome, because as I said, I could only replicate it once.

**Comment by DebuffPvP** [comment]
wait what if i buy a server host give you the info and put my files in there so then you can also use a vpn so that your ip is safe and you won't need to download any files either cause i promise it will happen on my server 100%

**Comment by OmerBenGera** [comment]
> 
> 
> wait what if i buy a server host give you the info and put my files in there so then you can also use a vpn so that your ip is safe and you won't need to download any files either cause i promise it will happen on my server 100%

It happens with only SSB and SGP, right?
No other plugins are installed, right?

**Comment by OmerBenGera** [comment]
Check out latest dev build, hopefully the issue is fixed

**Comment by DebuffPvP** [comment]
I think my friend experienced it a couple times when shop wasn't installed and then i myself did like twice but when shopguiplus is installed it happens every time and looks like the last dev build didn't fix it

**Comment by OmerBenGera** [comment]
> 
> 
> I think my friend experienced it a couple times when shop wasn't installed and then i myself did like twice but when shopguiplus is installed it happens every time and looks like the last dev build didn't fix it

Are you sure you tested it with latest dev build, fresh install?

**Comment by OmerBenGera** [comment]
fresh island etc?

**Comment by OmerBenGera** [comment]
also, when you restart, do you make sure the worth bonus and level bonus are saved correctly?

**Comment by OmerBenGera** [comment]
I mean, checking if they are in the database when the server is not running

**Comment by DebuffPvP** [comment]
can you tell me a step with procedures you want me to do in order to see if it occurs i'm not getting what you're asking

**Comment by DebuffPvP** [comment]
cause i did a fresh install and it still happens

**Comment by DebuffPvP** [comment]
wait so by restart which files should i delete

**Comment by DebuffPvP** [comment]
i think i found the issue i'm still testing but it's fixable

**Comment by DebuffPvP** [comment]
Ok so i don't really know how to explain it now that i see what it is but set the provider to buy in the config.yml and set the worths and levels to what i gave you on here https://pastebin.com/YpfLuwAj
before starting the server delete your db file then go and join the server after you start, create an island and place a pig spawner for example and then break it, then stop the server and start again and do /is top

**Comment by DebuffPvP** [comment]
from this point i'm clueless as to what the issue is if it's actually those causing it

**Comment by OmerBenGera** [comment]
I will replicate it step by step as you described it and will send a message here with what I got.

**Comment by DebuffPvP** [comment]
btw idk if u read what i said before but i can give you money to purchase a server if you want on pebblehost and i'll upload the stuff on it and then you can test my server

**Comment by OmerBenGera** [comment]
> 
> 
> btw idk if u read what i said before but i can give you money to purchase a server if you want on pebblehost and i'll upload the stuff on it and then you can test my server

there's no need for it as of now
If it won't work, I'll just download a zip file from you with all the server files.

**Comment by DebuffPvP** [comment]
you want me to make a zip?

**Comment by OmerBenGera** [comment]
> 
> 
> you want me to make a zip?

Not yet.
I will let you know if I need anything else from you.

**Comment by OmerBenGera** [comment]
Please send me your server as zipped
Make sure you have only necessary plugins for the issue to occur

**Comment by DebuffPvP** [comment]
https://mega.nz/file/BxoV1IhD#djvZTmsYjecm9_tkE34jP7SWhlXpu42GTajGoL6JFEc
Sorry it took so long i had to go to my mom's one of her patients was having a medical emergency

**Comment by DebuffPvP** [comment]
btw i'm using hyperion a taco spigot 1.8.8 fork

**Comment by OmerBenGera** [comment]
Check out latest dev build, hopefully it's fixed

**Comment by DebuffPvP** [comment]
So could you replicate the issue? btw i'll try it later cause i'm working rn

**Comment by OmerBenGera** [comment]
> 
> 
> So could you replicate the issue? btw i'll try it later cause i'm working rn

I could replicate the issue, yes
There were issues with counting blocks in legacy versions and a few other issues regarding counting of blocks in schematics.

**Comment by DebuffPvP** [comment]
Thanks you officially fixed but i got another issue when i remove items from the warp editing menu for your own island i can remove all items in that menu with no problem but when i remove the enderpearl it causes an error to happen and doesn't allow me to remove it and i'm talking about this one in this gif https://gyazo.com/ad815375fc16ac873b63006b630c2b1f

**Comment by OmerBenGera** [comment]
> Thanks you officially fixed but i got another issue when i remove items from the warp editing menu for your own island i can remove all items in that menu with no problem but when i remove the enderpearl it causes an error to happen and doesn't allow me to remove it and i'm talking about this one in this gif https://gyazo.com/ad815375fc16ac873b63006b630c2b1f

Open a new issue, and give more info as I dont quite understand what you mean.
